vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/DatabaseBackedProcessor.java

```diff
@@ -1163,18 +1163,38 @@ protected final <T> T submit(
                                                                             output.name()))))
                                                     .collect(Collectors.toSet());
 
-                                            if (externalIds.size() != externalKeys.size()
-                                                || requiredOutputKeys.size() != externalKeys.size()
-                                                || !requiredOutputKeys.equals(externalKeyIds)
-                                                || !externalKeyIds.containsAll(optionalOutputKeys)
-                                                || !externalKeyIds.equals(
-                                                    externalIds.stream()
-                                                        .map(
-                                                            k ->
-                                                                new Pair<>(
-                                                                    k.getProvider(), k.getId()))
-                                                        .collect(Collectors.toSet()))) {
-                                              return handler.externalIdMismatch();
+                                            if (externalIds.size() != externalKeys.size()) {
+                                              return handler.externalIdMismatch(String.format(
+                                                  "%d External IDs found (%s) but %d External Keys found (%s)!",
+                                                  externalIds.size(),
+                                                  externalIds, externalKeys.size(), externalKeys));
```

Comment:
Since it's a set should it be `externalIds.toString()` ?

Reply:
Nope, Sets will write themselves out without prompting. Intellij actually advises 'remove redundant toString()' if you try